install/install.php

```diff
@@ -112,16 +87,16 @@ public static function get_default_options() {
 	 * @return void
 	 */
 	protected function _activate() {
-		if ( $options = get_option( 'polylang' ) ) {
-			// Check if we will be able to upgrade
+		$options = $this->get_options();
+
+		if ( ! empty( $options['version'] ) ) {
+			// Check if we will be able to upgrade.
 			if ( version_compare( $options['version'], POLYLANG_VERSION, '<' ) ) {
-				$upgrade = new PLL_Upgrade( $options );
-				$upgrade->can_activate();
+				( new PLL_Upgrade( $options ) )->can_activate();
 			}
-		}
-		// Defines default values for options in case this is the first installation
-		else {
-			update_option( 'polylang', self::get_default_options() );
+		} else {
+			$options['version'] = POLYLANG_VERSION;
+			$this->save_options( $options );
```

Comment:
Because [I removed the hook on `shutdown`](https://github.com/polylang/polylang/pull/1481/files#diff-355427764f2ae86e58c72233eb32272eeb381e9ade0c017d4d0f4f8c28ce353fR152) to not interfere with another instance of `Options`.
Maybe there is better way to deal with all this.

The pain point is that during the activation hook there is no existing instance of `Options`, while there is one during the upgrade process.

Reply:
Changed in [ea0bcd3](https://github.com/polylang/polylang/pull/1481/commits/ea0bcd34fc5e6ef6af5e96d23dee4277a39fa10d).